grass7/vector/v.centerline/v.centerline.py

```diff
@@ -108,8 +108,8 @@ def main():
     categories = grass.pipe_command('v.category', input=input, option='print',
             quiet=True)
     for category in categories.stdout:
-        segment_input += 'P {}'.format(category.strip())
-        segment_input += ' {} {}'.format(category.strip(),' 50%')
+        segment_input += 'P {}'.format(category.strip().encode('utf8'))
+        segment_input += ' {} {}'.format(category.strip().encode('utf8'),' 50%')
```

Comment:
First, it's not clear to me why this uses pipe_command, seems read_command would be sufficient. With read_command, the output will be decoded, so everything should work. If you keep pipe_command, then you need to call grass.decode(categories.stdout), to get string from bytes.

Reply:
Maybe @mlennert can give some hints why `pipe_command()` has been used?

Reply:
This was written quite quickly some time ago and since I didn't follow good practice in terms of commenting, I'm not sure why I used this approach here. IIRC it was a general attempt at the time to avoid any potential memory issues by not creating too many intermediate data structures. The same is true for the pipe_command call to v.distance.

I don't remember if I had any cases where this was actually an issue or whether this was just me trying to be elegant and prevent potential future problems. Possibly overkill in this particular case and in this day and time of abundance in RAM.